packages/core/src/routes/logto-config.ts

```diff
@@ -287,4 +290,57 @@ export default function logtoConfigRoutes<T extends AuthedRouter>(
       return next();
     }
   );
+
+  if (!EnvSet.values.isCloud) {
+    return;
+  }
+
+  router.post(
+    '/configs/jwt-customizer/test',
+    koaGuard({
+      body: z.discriminatedUnion('tokenType', [
+        z.object({
+          tokenType: z.literal(LogtoJwtTokenKey.AccessToken),
+          payload: accessTokenJwtCustomizerGuard,
+        }),
+        z.object({
+          tokenType: z.literal(LogtoJwtTokenKey.ClientCredentials),
+          payload: clientCredentialsJwtCustomizerGuard,
+        }),
+      ]),
+      response: jsonObjectGuard,
+      /**
+       * TODO (LOG-8450): Add this path to openapi.json file, since now the swagger.json can not automatically adapt the path validation for OSS version.
+       *
+       * Code 400 indicates Zod errors in cloud service (data type does not match expectation, can be either request body or response body).
+       * Code 422 indicates syntax errors in cloud service.
+       * Code 500 indicates internal server errors in cloud service.
+       */
+      status: [200, 400, 422, 500],
+    }),
+    async (ctx, next) => {
+      const {
+        body: {
+          payload: { tokenSample, contextSample, ...rest },
+        },
+      } = ctx.guard;
+
+      /**
+       * We have ensured the API request body via koa guard, manually cast the cloud service API call's
+       * `requestBody` type and let the cloud service API to throw if needed.
+       */
+      // eslint-disable-next-line no-restricted-syntax, @typescript-eslint/consistent-type-assertions
+      const requestBody = {
+        ...rest,
+        token: tokenSample,
+        context: contextSample,
+      } as CustomJwtFetcher;
```

Comment:
1. update the `customJwtFetcherGuard` token field to `z.record<z.unknown()>`;
2. align the optional field of the test API with the service API. like script and token